src/DependencyInjection/Compiler/PluginsPass.php

```diff
@@ -29,15 +31,40 @@ public function process(ContainerBuilder $container)
             foreach ($buses as $name => $bus) {
                 $globalPlugins = $container->findTaggedServiceIds('prooph_service_bus.plugin');
                 $typePlugins = $container->findTaggedServiceIds(sprintf('prooph_service_bus.%s.plugin', $type . '_bus'));
-                $plugins = $container->findTaggedServiceIds(sprintf('prooph_service_bus.%s.plugin', $name));
+                $localPlugins = $container->findTaggedServiceIds(sprintf('prooph_service_bus.%s.plugin', $name));
 
-                $plugins = array_merge(array_keys($globalPlugins), array_keys($typePlugins), array_keys($plugins));
+                $plugins = array_merge(array_keys($globalPlugins), array_keys($typePlugins), array_keys($localPlugins));
 
                 $busDefinition = $container->getDefinition($bus);
-                $busPlugins = $busDefinition->getArgument(2);
 
-                $finalPlugins = array_merge($busPlugins, $plugins);
-                $busDefinition->replaceArgument(2, $finalPlugins);
+                foreach ($plugins as $plugin) {
+                    $busDefinition->addMethodCall('addPlugin', [new Reference($plugin), $plugin]);
+                }
+
+
+                $resolveClass = function ($id) use ($container): string {
```

Comment:
This can be outside of the loop, the $container is reference.

In fact you can also make this a private method and keep the container as property, but as this method is only called once (during container building) this will not make a big difference.